### DISCUSSION TITLE
Code blocks are ignoring language syntax highlighting and newlines

### DISCUSSION THREAD

**Discussion by rkyathappala** [facelessuser/MarkdownPreview#179] [Q&A] [answered]
I just installed on ST3 today, havent looked at all documentation/issues. but after working a while to get the mathjax equation support to work I noticed that the code  block is ignoring the syntax highlighting for my basic c++ block, as well as ignoring the new lines in the code i wrote.

I appreciate the project, and thanks for the support!

My example is [Test-subl.md](https://github.com/facelessuser/MarkdownPreview/files/14733272/Test-subl.md) which yields:
![image](https://github.com/facelessuser/MarkdownPreview/assets/16108635/89e58f2a-913f-4d10-adb8-b0627ee04cfc)

**Comment by facelessuser** [comment]
Please provide your settings.

**Comment by rkyathappala** [comment]
```json
{
	/*
		Enable mathjax support for LaTeX formulas

        To keep other extensions active, you can specify in the place of //etc..
	*/
	"enable_mathjax": true,
	"js": [
        "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML",
        "res://MarkdownPreview/js/math_config.js",
    ],
    "markdown_extensions": [
        "markdown.extensions.extra",
        {
            "pymdownx.arithmatex": {
                "generic": true
            }
        },
        {
            "markdown.extensions.toc": {
                "permalink": "\ue157"
            }
        }
        //---- etc.
    ]
}

```

**Comment by rkyathappala** [comment]
> Please provide your settings.

```json
{
	/*
		Enable mathjax support for LaTeX formulas

                To keep other extensions active, you can specify in the place of //etc..
	*/
	"enable_mathjax": true,
	"js": [
          "https://cdn.mathjax.org/mathjax/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML",
          "res://MarkdownPreview/js/math_config.js",
        ],
        "markdown_extensions": [
          "markdown.extensions.extra",
          {
              "pymdownx.arithmatex": {
                  "generic": true
              }
          },
          {
              "markdown.extensions.toc": {
                  "permalink": "\ue157"
              }
          }
          //---- etc.
        ]
}
```

**Comment by facelessuser** [comment, marked as answer]
`//--- etc.` doesn't help me see what is wrong, so let me say this, when you override plugins in your personal settings, you override _all_ plugins, so make sure you include any and all that you wish to use. If you want syntax highlighting in blocks, you need to have `pymdownx.superfences` in your plugin list. Look at the default settings and copy them over to your user settings.

**Comment by rkyathappala** [comment]
> `//--- etc.` doesn't help me see what is wrong, so let me say this, when you override plugins in your personal settings, you override _all_ plugins, so make sure you include any and all that you wish to use. If you want syntax highlighting in blocks, you need to have `pymdownx.superfences` in your plugin list. Look at the default settings and copy them over to your user settings.

Yep that fixed it, thanks!

I had kept the //etc in there from the mathjax thread I was following on stackOverflow. 

In the future if I find a specific md feature not working as I expect it where should I look to figure out the specific md extension name so I can copy it over to my user setttings?

Or would it just be better practice to copy over all the md extensions in the default settings?

**Comment by facelessuser** [comment]
> Or would it just be better practice to copy over all the md extensions in the default settings?

I would copy everything over and only emove the things you don't want.
